### PR TITLE
Skip hidden files when counting visible files in list

### LINKED ISSUE #26518: No content is shown on folders with many hidden Folders
### Steps to reproduce
1. Create a folder with <30 of hidden Folders inside, and one other file
2. Open the Folder

### Expected behaviour
The normal file is shown

### Actual behaviour
No Files are shown, only the stats line appear. If the browser window is small enough to show a scroll bar the file is appearing after scrolling down. This seems to be an issue with the Paged load. Toggling the Show Hidden Files Option Resolve it too.

owncloud

### Server configuration
**Operating system**: Debian Jessie

**Web server:** Nginx 1.9.10

**Database:** Postgres 9.4

**PHP version:** 5.6.27

**ownCloud version:** 9.1.1 + patches from #26399 and #26517

**Updated from an older ownCloud or fresh install:** Upgraded from 8.2 (via 9.0)

**Where did you install ownCloud from:** Owncloud deb repository

**Signing status (ownCloud 9.0 and above):** unkown

```
No errors have been found.
```

**List of activated apps:**

```
Enabled: 
    - activity: 2.3.2 
    - comments: 0.3.0 
    - dav: 0.2.6 
    - external: 1.2 
    - federatedfilesharing: 0.3.0 
    - federation: 0.1.0 
    - files: 1.5.1 
    - files_external: 0.6.0 
    - files_pdfviewer: 0.8.1 
    - files_sharing: 0.10.0 
    - files_texteditor: 2.1 
    - files_trashbin: 0.9.0 
    - files_versions: 1.3.0 
    - files_videoplayer: 0.9.8 
    - gallery: 15.0.0 
    - notifications: 0.3.0 
    - provisioning_api: 0.5.0 
    - systemtags: 0.3.0 
    - templateeditor: 0.1 
    - updatenotification: 0.2.1 
    - user_ldap: 0.9.0 Disabled: 
    - encryption 
    - files_antivirus 
    - firstrunwizard 
    - user_external
```

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "*** REMOVED ***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [  *** REMOVED *** ],
        "datadirectory": "\/opt\/storage\/data",
        "dbtype": "pgsql",
        "version": "9.1.1.3",
        "dbname": "owncloud",
        "dbhost": "*** REMOVED ***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "forcessl": false,
        "ldapIgnoreNamingRules": false,
        "theme": "mafiasi",
        "mail_smtpmode": "smtp",
        "mail_smtphost": "*** REMOVED ***",
        "mail_from_address": "*** REMOVED ***",
        "mail_domain": "*** REMOVED ***",
        "maintenance": false,
        "minimum.supported.desktop.version": "1.5",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "filelocking.enabled": "true",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "localhost",
            "port": 6379,
            "timeout": 0,
            "password": "***REMOVED SENSITIVE VALUE***"
        },
        "trashbin_retention_obligation": "auto",
        "overwrite.cli.url": "*** REMOVED ***",
        "logtimezone": "UTC",
        "logfile": "\/opt\/log\/owncloud.log",
        "loglevel": 1,
        "updatechecker": false
    }
}
```

**Are you using external storage, if yes which one:** only the primary local storage

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP

#### LDAP configuration (delete this part if not used)

```
+-------------------------------+---------------------------------------------+
| Configuration                 |                                             |
+-------------------------------+---------------------------------------------+
| hasMemberOfFilterSupport      |                                             |
| hasPagedResultSupport         |                                             |
| homeFolderNamingRule          |                                             |
| lastJpegPhotoLookup           | 0                                           |
| ldapAgentName                 | cn=*** REMOVED ***,dc=*** REMOVED ***,dc=de |
| ldapAgentPassword             | ***                                         |
| ldapAttributesForGroupSearch  | cn                                          |
| ldapAttributesForUserSearch   | uid;sn;displayName;givenName                |
| ldapBackupHost                |                                             |
| ldapBackupPort                |                                             |
| ldapBase                      | dc=*** REMOVED ***,dc=de                    |
| ldapBaseGroups                | ou=groups,dc=*** REMOVED ***,dc=de          |
| ldapBaseUsers                 | ou=People,dc=*** REMOVED ***,dc=de          |
| ldapCacheTTL                  | 600                                         |
| ldapConfigurationActive       | 1                                           |
| ldapDynamicGroupMemberURL     |                                             |
| ldapEmailAttribute            | mail                                        |
| ldapExperiencedAdmin          | 0                                           |
| ldapExpertUUIDGroupAttr       |                                             |
| ldapExpertUUIDUserAttr        |                                             |
| ldapExpertUsernameAttr        | uid                                         |
| ldapGroupDisplayName          | cn                                          |
| ldapGroupFilter               | (&(|(objectclass=posixGroup)))              |
| ldapGroupFilterGroups         |                                             |
| ldapGroupFilterMode           | 0                                           |
| ldapGroupFilterObjectclass    | posixGroup                                  |
| ldapGroupMemberAssocAttr      | memberUid                                   |
| ldapHost                      | ldap://*** REMOVED ***                      |
| ldapIgnoreNamingRules         |                                             |
| ldapLoginFilter               | (&(|(objectclass=inetOrgPerson))(uid=%uid)) |
| ldapLoginFilterAttributes     |                                             |
| ldapLoginFilterEmail          | 0                                           |
| ldapLoginFilterMode           | 1                                           |
| ldapLoginFilterUsername       | 1                                           |
| ldapNestedGroups              | 0                                           |
| ldapOverrideMainServer        | 0                                           |
| ldapPagingSize                | 500                                         |
| ldapPort                      | 389                                         |
| ldapQuotaAttribute            | ownCloudQuota                               |
| ldapQuotaDefault              | 2 GB                                        |
| ldapTLS                       | 0                                           |
| ldapUserDisplayName           | displayname                                 |
| ldapUserDisplayName2          |                                             |
| ldapUserFilter                | (|(objectclass=inetOrgPerson))              |
| ldapUserFilterGroups          |                                             |
| ldapUserFilterMode            | 0                                           |
| ldapUserFilterObjectclass     | inetOrgPerson                               |
| ldapUuidGroupAttribute        | auto                                        |
| ldapUuidUserAttribute         | auto                                        |
| turnOffCertCheck              | 0                                           |
| useMemberOfToDetectMembership | 1                                           |
+-------------------------------+---------------------------------------------+
```

### Client configuration
**Browser:** Firefox 47

**Operating system:** Debian Testing

#### Web server error log
```
No log available
```

#### ownCloud log (data/owncloud.log)
```
No Errors regarding this problem
```

#### Browser log
```
No JS errors
```

### PR DESCRIPTION
**Pull request by PVince81**
## Description
The file list UI logic that renders the next page doesn't properly
exclude hidden files when not shown. This fix makes sure that only rows
that are actually visible are counted when rendering a page, this makes
sure that the page will always have enough elements displayed.

## Related Issue
Fixes https://github.com/owncloud/core/issues/26518

## How Has This Been Tested?
See steps in original ticket.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Backports:
- [x] stable9.1

Please review @jvillafanez @PhilippSchaffrath @DeepDiver1975 @felixheidecke

### CODE REVIEW

**Comment by davitol** [comment]
:+1: WFM

File: apps/files/js/filelist.js [hunk 1]
```diff
@@ -938,7 +939,10 @@
 				}
 				newTrs.push(tr);
 				index++;
-				count--;
+				// only count visible rows
```
Comment (jvillafanez) [inline apps/files/js/filelist.js]: A clearer comment might be needed here. This feels weird.

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by PVince81** [comment]
stable9.1: https://github.com/owncloud/core/pull/26571

**Pull request merged**